_posts/en/newsletters/2020-08-12-newsletter.md

```diff
@@ -0,0 +1,215 @@
+question below to see a summary of the answer from the meeting.*
+
+[Implement ADDRv2 support (part of BIP155)][review club #19031] is a PR
+([#19031][Bitcoin Core #19031]) by Vasil Dimov that proposes an
+implementation of the [BIP155][] `addrv2` P2P message.
+
+The discussion covered the current `addr` message in Bitcoin Core, the BIP155
+`addrv2` message, how to signal support for `addrv2` on the P2P network, and the
+necessity of `addrv2` for the Bitcoin network to upgrade to Tor v3. Tor v2
+deprecation begins mid-September!
+
+{% include functions/details-list.md
+  q0="How does Bitcoin Core handle network addresses?"
+  a0="Bitcoin Core currently sees all peer addresses as being more or less in a
+      [single][19031 single] IPv6-like space. IPv4 and Tor v2 addresses are
+      serialized into [16 bytes][19031 fake] encoded as “fake” IPv6 addresses
+      from specific IPv6 networks reserved for them. For instance, Tor v2
+      addresses are persisted using the [“onioncat” IPv6 range][19031 onioncat].
+      The 16-byte maximum length of IPv6 increasingly limits which endpoints
+      Bitcoin nodes can connect to."
+  a0link="https://bitcoincore.reviews/19031#l-35"
+
+  q1="How are peer addresses stored?"
+  a1="Addresses are saved in the `peers.dat` file in a 16-byte format, which
+      corresponds to the length of IPv6 and Tor v2 addresses; the smaller-sized
+      IPv4 addresses are therefore [padded][19031 padded] to fit. This file can
+      be updated to handle larger addresses with backward compatibility
+      maintained for older versions."
+  a1link="https://bitcoincore.reviews/19031#l-82"
+
+  q2="What is BIP155 and the `addrv2` message?"
+  a2="[BIP155 `addrv2`][BIP155] is a new P2P message format proposed in early
+      2019 by Wladimir J. van der Laan to gossip variable-length node addresses
+      larger than 16 bytes, and up to 512 bytes for future extensibility, with
+      separate address spaces for separate networks. `addrv2` would enable
+      Bitcoin core to use [next-generation Onion Services][19031 nextgen]
+      (Tor v3) and the [I2P (Invisible Internet Project)][19031 i2p], as well as
+      other networks with longer endpoint addresses that don't fit in
+      the 16 bytes/128 bits of Bitcoin's current `addr` message."
+
+  q3="How is `addrv2` support expected to be signaled?"
+  a3="Most likely with a `sendaddrv2` message between peers
+      [at handshake or afterwards][19031 handshake], rather than with the
+      originally proposed protocol version bump or with a new network service
+      bit."
+  a3link="https://bitcoincore.reviews/19031#l-99"
+
+  q4="Why is Tor v3 important for Bitcoin?"
+  a4="Tor v2 is planned to be [deprecated on Sept 15, 2020][19031 v2 deprecate]
+      and obsolete on July 15, 2021 ([schedule][19031 v2 schedule]). Tor v3
+      provides much better security and privacy. It uses longer, 56-character
+      addresses and therefore requires implementation of BIP155 `addrv2`."
```

Comment:
> "Tor v3 provides much better security and privacy, since it uses longer, 56-character addresses. Using Tor v3 on the Bitcoin network therefore requires implementation of BIP155 `addrv2`."

Much better :+1: